mssql_conditions.py

```diff
@@ -47,9 +32,7 @@ def get_where_conditions(self, signal, table, cursor):
             else:
                 conditions += ' {} '.format(combine_condition)
 
-            condition_string = '{} {} ?'.format(
-                self.validate_column(condition.column(signal), table, cursor),
-                condition.operation(signal).value)
+            condition_string = '{} {} ?'.format(self.validate_column(condition.column(signal), table, cursor), condition.operation(signal).value)
```

Comment:
i know this was here before, but this string formatting stuff makes me _really_ uncomfortable, especially when it is derived from incoming signals. I, literally, cannot express how uncomfortable it makes me.

Reply:
Actually, I parsed this code a little closer, and it's not as bad as I thought it was, most of these blocks get pushed through `get_where_conditions()` which will call `validate_column()`, which at least checks that the column exists on the table. And then uses parametric parameter injection. Unfortunately, the `table` names are injected directly, and evaulated from the signal. So `{ "table": "table; -- DROP TABLE Users" }` is going to be less than ideal. It probably wouldn't be outrageous to do the same column-check on tables as well—or treat table like a static/block property rather than a signal-property, so it's at least not injectable. _I.e._ if you screw up writing the expression in your _service_ then thats on you.

However, there one block that _does_ still seem dangerous is the InsertBlock. That one doesn't seem to do any validation and is all string injection, with its _own_ string escaping! Yikes.